### PR TITLE
Add support for new python versions and update pkg metadata

### PR DESCRIPTION
**Pull request by mtreinish**
This commit updates the supported python versions for subunit to include
support for newer versions of the language 3.7 and 3.8 which have been
released since the last python-subunit release. At the same time it
takes the opportunity to update the package metadata to add more details
including the new python versions.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -53,14 +53,25 @@ def _get_version_from_file(filename, start_of_line, split_marker):
     long_description=open('README.rst').read(),
     classifiers=[
         'Intended Audience :: Developers',
+        'Operating System :: OS Independent',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
```
Comment (tosky) [inline setup.py]: This seems duplicate (see line 57 above), and maybe (just for ordering reasons) the 2.7 classifier should be moved before the :: 3 classifier.

Comment (mtreinish) [reply setup.py]: Ah, good call. I'll fix this one sec

**Comment by tosky** [comment]
Thanks, looks fine (at least from my user point of view).

**Comment by FedericoRessi** [comment]
I would create a new release tag before releasing this change and use this one as an
the opportunity to remove Python 2.7 and Python 3.4 support too for the following incoming releases.
These would be the two released to be created in the near future:
- 1.4.0 => last release compatible with Python 2.7, Python 3.4-3.6
- 1.5.0 => first release compatible with Python 3.5 - 3.8

**Pull request merged**